python/cudf/cudf/core/index.py

```diff
@@ -1856,7 +1849,7 @@ def __repr__(self):
     @copy_docstring(StringMethods.__init__)
     @property
     def str(self):
-        return self._values.str(parent=self)
+        return StringMethods(column=self._values, parent=self)
```

Comment:
We should probably make this a cached properly while we're at it as well

Reply:
Don't need to tackle this in this PR, maybe just file an issue for future enhancement

Reply:
#5695